src/m365/pa/commands/app/app-permission-list.ts

```diff
@@ -0,0 +1,125 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { formatting } from '../../../../utils/formatting';
+import { odata } from '../../../../utils/odata';
+import { validation } from '../../../../utils/validation';
+import PowerAppsCommand from '../../../base/PowerAppsCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  appName: string;
+  asAdmin?: boolean;
+  environmentName?: string;
+  roleName?: string;
+}
+
+class PaAppPermissionListCommand extends PowerAppsCommand {
+  private readonly allowedRoleNames = ['Owner', 'CanEdit', 'CanView'];
+
+  public get name(): string {
+    return commands.APP_PERMISSION_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all permissions of a Power Apps app';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['roleName', 'principalId', 'principalType'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        asAdmin: !!args.options.asAdmin,
+        environmentName: typeof args.options.environment !== 'undefined',
+        roleName: typeof args.options.roleName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--appName <appName>'
+      },
+      {
+        option: '--asAdmin'
+      },
+      {
+        option: '-e, --environmentName [environmentName]'
+      },
+      {
+        option: '--roleName [roleName]',
+        autocomplete: this.allowedRoleNames
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.appName)) {
+          return `${args.options.appName} is not a valid GUID.`;
```

Comment:
I'm being more and more of a fan of showing the value + option name in the error. When using this in a script with variables as option values, it's much easier to find the bug.

```suggestion
          return `${args.options.appName} is not a valid GUID for appName.`;
```